drivers/ethernet/eth_xlnx_gem_priv.h

```diff
@@ -511,11 +518,28 @@ struct eth_xlnx_dma_area_gem##port {\
 		& ~(ETH_XLNX_BUFFER_ALIGNMENT - 1))];\
 };
 
-/* DMA memory area instantiation macro */
+#if defined(CONFIG_SOC_XILINX_ZYNQMP)
+#define ETH_XLNX_GEM_DMA_AREA_Q1_DECL(port) \
+struct eth_xlnx_dma_q1_area_gem##port {\
+	struct eth_xlnx_gem_bd tx_bd[1];\
```

Comment:
Hi @yashi,
sorry for the long reply delay.

> Not sure what you are trying to convey. I assume that what you are trying to say with this patch is to fix the lockup and doesn't prevent anyone from utilizing the 2nd set of DMA buffers.

That's correct, the immediate purpose is to fix the lockup by adding a single, always idle, TX buffer descriptor but no RX buffer descriptor, as this direction is not affected by the problem. Yet, the 2nd set of RX/TX buffer descriptors /could/ be made fully functional at a later time.

At the time being, from my point of view it's favourable to stick with the original set of RX/TX BD descriptor rings for actual reception and transmission of packets, pointed to the by rx_qbar (@ GEM base + 0x18) and tx_qbar (@ GEM base + 0x1C) registers. This way, the effective RX/TX-related code stays the same for the Zynq-7000 (which doesn't have the 2nd set of registers introduced with the UltraScale), and the UltraScale fix is just this small addition on top which adds this one dummy TX descriptor and lets the 2nd TX queue base address register point to it.

/However/, this would have to change if someone started using the GEM not from the context of the UltraScale's 32-bit R5 cores, but from the 64-bit A53 cores and wanted to place the respective GEM's DMA area in a memory area located above 4GB - it's only the 2nd, UltraScale-only set of registers with their additional high address word registers which enables the placement of RX/TX descriptor rings in memory above 4GB. In that case, the accordingly enlarged buffer descriptor format with two address words would have to be supported. If 64-bit support was to be added, there'd either be a significant amount of relatively similar code #ifdef'd to the respective SoC family, or maybe the driver sources should be split into two variants for improved readability. But as the A53 cores aren't yet relevant to my work, I haven't experimented in this direction yet.